lib/MusicBrainz/Server/Data/Search.pm

```diff
@@ -594,6 +600,7 @@ sub schema_fixup
             # The search server returns the MBID in the 'id' attribute, so we
             # need to rename that.
             $entity{gid} = delete $entity{id};
+            %entity = %{ $self->schema_fixup_mapping(\%entity) };
             %entity = %{ $self->schema_fixup_type(\%entity, $entity_type) };
```

Comment:
I *think* you can just replace the `schema_fixup_type` line here with a call to `schema_fixup`, and revert everything else. That should both fix the mapping and call `schema_fixup_type` (and perhaps do other things we've forgotten, so it seems somewhat preferable).